blueprints/reserved-capacity/README.md

```diff
@@ -0,0 +1,280 @@
+
+### Cleanup
+
+```sh
+kubectl delete -f reserved-instances.yaml -f workload-reserved.yaml
+```
+
+## Scenario 3: On-Demand Capacity Reservations (ODCRs)
+
+If you have active ODCRs and want to use native ODCR support:
+
+**Scenario:** You need guaranteed capacity for critical workloads during peak traffic or launch events. You've created On-Demand Capacity Reservations to ensure instances are always available when needed. You want Karpenter to automatically use these reserved instances first, with seamless fallback to regular on-demand when reservations are full.
+
+### Configuration
+
+```yaml
+apiVersion: karpenter.k8s.aws/v1
+kind: EC2NodeClass
+metadata:
+  name: odcr-nodeclass
+spec:
+  capacityReservationSelectorTerms:
+  - tags:
+      intent: apps
+  # Or select by specific ODCR ID:
+  # - id: cr-1234567890abcdef0
+---
+apiVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  name: odcr-priority
+spec:
+  template:
+    metadata:
+      labels:
+        intent: apps
+    spec:
+      nodeClassRef:
+        name: odcr-nodeclass
+      requirements:
+      - key: karpenter.sh/capacity-type
+        operator: In
+        values: ["reserved", "on-demand"]  # Reserved first, on-demand fallback
+```
+
+### Deploy
+
+If you're using the Terraform template provided in this repo, run the following commands to get the EKS cluster name and the IAM Role name for the Karpenter nodes:
+
+```sh
+export CLUSTER_NAME=$(terraform -chdir="../../cluster/terraform" output -raw cluster_name)
+export KARPENTER_NODE_IAM_ROLE_NAME=$(terraform -chdir="../../cluster/terraform" output -raw node_instance_role_name)
+```
+
+Then deploy the ODCR scenario:
+
+```sh
+sed -i '' "s/<<CLUSTER_NAME>>/$CLUSTER_NAME/g" odcr.yaml
+sed -i '' "s/<<KARPENTER_NODE_IAM_ROLE_NAME>>/$KARPENTER_NODE_IAM_ROLE_NAME/g" odcr.yaml
+kubectl apply -f odcr.yaml
+kubectl apply -f workload-odcr.yaml
+```
+
+### Results
+
+**Check the results:**
+```sh
+kubectl get nodes -L karpenter.sh/capacity-type,karpenter.k8s.aws/capacity-reservation-id
+```
+
+**Expected output:**
+```
+NAME                                        STATUS   CAPACITY-TYPE   CAPACITY-RESERVATION-ID
+ip-10-0-x-x.region.compute.internal         Ready    on-demand       
+ip-10-0-x-x.region.compute.internal         Ready    on-demand       
+```
+
+```sh
+# Check Karpenter logs to see reserved capacity attempted first
+kubectl logs -n karpenter -l app.kubernetes.io/name=karpenter --tail=50 | grep -E "replacement-nodes.*reserved|launched.*capacity-type"
+```
+
+**Expected output:**
+```
+"replacement-nodes":[{"capacity-type":"reserved",...}]  # Planned reserved first
+"launched nodeclaim"..."capacity-type":"on-demand"     # Fell back to on-demand
+```
+
+**What this demonstrates:**
```

Comment:
I love this approach you're taking in this blueprint, well done ... one thing I think it's pretty important to highlight is that this configuration is simpler than the others as you don't have to have extra NodePools or think about the weights to use. Can you add that highlight?

Also, it's important to say that WE DO recommend customers using this approach. Specially as the guidance now is to use Saving Plans, and with Saving Plans you're not reserving capacity. You can refer readers to this docs: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-capacity-reservations.html#capacity-reservations-differences